src/App.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from "react";
+import "bulma/css/bulma.min.css";
+import "./App.css";
+
+function App() {
+    const INIT_SIZE = 3;
+
+    const initGrid = (size) => {
+        const newGrid = [];
+        for (let x = 0; x < size; x++) {
+            const arr = new Array(size);
+            arr.fill({});
+            newGrid.push(arr);
+        }
+        return newGrid;
+    };
+
+    const initial = {
+        size: INIT_SIZE,
+        grid: initGrid(INIT_SIZE),
+    };
+    const [state, setState] = useState(initial);
+
+    const handleChange = (e) => {
+        let newSize = e.target.value;
+        newSize = parseInt(newSize, 10) > 40 ? 40 : parseInt(newSize, 10);
+        const newGrid = initGrid(newSize);
+        setState({ ...state, size: newSize, grid: newGrid });
+    };
+    const toggleClass = (e) => {
+        e.target.classList.toggle("active");
+    };
+    return (
```

Comment:
Think this could be broken down into React components so there's not so many nested divs/tags

Reply:
avoid the wrapper hell!!